src/synonym_map.c

```diff
@@ -127,13 +127,15 @@ uint32_t SynonymMap_Add(SynonymMap* smap, const char** synonyms, size_t size) {
 
 void SynonymMap_Update(SynonymMap* smap, const char** synonyms, size_t size, uint32_t id) {
   RS_LOG_ASSERT(!smap->is_read_only, "SynonymMap should not be read only");
+  char *syn_str[size];
   int ret;
   for (size_t i = 0; i < size; i++) {
+    syn_str[i] = strtolower(rm_strdup(synonyms[i]));
```

Comment:
Why not just define it as a temporary char inside the for?

Reply:
@MeirShpilraien 
ASAN detects a memory leak here that was exposed once the `strdup` function was moved from within TermData_New in `line 136`.
Freeing when index is dropped seems correct. Still investigating.

Reply:
fixed